http3/server.go

```diff
@@ -530,7 +548,8 @@ func ListenAndServe(addr, certFile, keyFile string, handler http.Handler) error
 	}
 
 	quicServer := &Server{
-		Server: httpServer,
+		Server:           httpServer,
+		customAltSvcPort: customAltSvcPort,
```

Comment:
Another possible way to do this would be to just set `port` to `customAltSvcPort` here, and then `SetQuicHeaders` would find the port is already having a non-zero value.  Open to suggestions of what you think is the best way to do this. :)